_posts/2018-05-10-chainermn-on-kubernetes.md

```diff
@@ -0,0 +1,209 @@
+
+## How to run ChainerMN on Kubernetes
+I explain it in three steps below:
+
+- [Step 1. Build Your Container Image](#step-1-build-your-docker-image)
+- [Step 2. Install Kubeflow's OpenMPI package](#step-2-install-kubeflows-openmpi-package)
+- [Step 3. Run ChainerMN on Kubernetes](step-3-run-chainermn-on-kubernetes)
+
+### Prerequisites
+- [Kubernetes](https://kubernetes.io/) cluster equipped with Nvidia GPUs
+- on your local machine
+  - [docker](https://www.docker.com/community-edition)
+  - [kubectl](https://kubernetes.io/docs/tasks/tools/install-kubectl/)
+  - [ksonnnet](https://ksonnet.io/)
+
+### Step 1. Build Your Container Image
+
+First we need to build a container image to run your deep learning workload with ChainerMN. All we can just follow [the official ChainerMN installation guides](http://chainermn.readthedocs.io/en/stable/installation/index.html).
+
+For [Chainer](https://chainer.org/)/[Cupy](https://cupy.chainer.org/), official docker image [`chainer/chainer`](https://hub.docker.com/r/chainer/chainer/) is available on DockerHub.  This is very handy as a base image or runtime image for deep learning workloads because this image is already `nvidia-docker` ready.
+
+Below is a sample `Dockerfile` to install CUDA aware [OpenMPI](https://www.open-mpi.org/), [NCCL](https://developer.nvidia.com/nccl), [ChainerMN](https://github.com/chainer/chainermn) and its sample `train_mnist.py` script.  Please save the contents with the name `Dockerfile`.
+
+```
+FROM chainer/chainer:v4.0.0-python3
+
+ARG OPENMPI_VERSION="2.1.3"
+ARG NCCL_PACKAGE_VERSION="2.1.4-1+cuda8.0"
+ARG CHAINER_MN_VERSION="1.2.0"
+
+# Install basic dependencies and locales
+RUN apt-get update && apt-get install -yq --no-install-recommends \
+      locales wget sudo ca-certificates ssh build-essential && \
+    rm -rf /var/lib/apt/lists/* /var/cache/apt/archives/* && \
+    echo "en_US.UTF-8 UTF-8" > /etc/locale.gen && locale-gen
+
+# Install OpenMPI with cuda
+RUN cd /tmp && \
+  wget -q https://www.open-mpi.org/software/ompi/v${OPENMPI_VERSION%\.*}/downloads/openmpi-$OPENMPI_VERSION.tar.bz2 && \
+  tar -xjf openmpi-$OPENMPI_VERSION.tar.bz2 && \
+  cd /tmp/openmpi-$OPENMPI_VERSION && \
+  ./configure --prefix=/usr --with-cuda && make -j2 && make install && rm -r /tmp/openmpi-$OPENMPI_VERSION* && \
+  ompi_info --parsable --all | grep -q "mpi_built_with_cuda_support:value:true"
+
+# Install NCCL2
+RUN wget http://developer.download.nvidia.com/compute/machine-learning/repos/ubuntu1604/x86_64/nvidia-machine-learning-repo-ubuntu1604_1.0.0-1_amd64.deb && \
+    dpkg -i nvidia-machine-learning-repo-ubuntu1604_1.0.0-1_amd64.deb && \
+    rm nvidia-machine-learning-repo-ubuntu1604_1.0.0-1_amd64.deb && \
+    apt-get update && \
+    apt-get install -yq --no-install-recommends \
+      libnccl-dev=${NCCL_PACKAGE_VERSION} libnccl2=${NCCL_PACKAGE_VERSION} && \
+  rm -rf /var/lib/apt/lists/* /var/cache/apt/archives/*
```

Comment:
Do you really need to install NCCL2 itself? When you install cupy with `pip install cupy-cuda80` command, it also brings built binaries of cuDNN, NCCL, and CUDA.